app/static/app/js/components/Map.jsx

```diff
@@ -184,6 +186,78 @@ class Map extends React.Component {
       secondaryAreaUnit: 'acres'
     });
     measureControl.addTo(this.map);
+    
+    const featureGroup = L.featureGroup();
+    featureGroup.addTo(this.map);
+
+    new L.Control.Draw({
+        draw: {
+          polygon: {
+            allowIntersection: false, // Restricts shapes to simple polygons
+            shapeOptions: {
+              color: '#707070'
+            }
+          },
+          rectangle: false,
+          circle: false,
+          circlemarker: false,
+          marker: false
+        },
+        edit: {
+          featureGroup: featureGroup,
+          edit: {
+            selectedPathOptions: {
+              maintainColor: true,
+              dashArray: '10, 10'
+            }
+          }
+        }
+      }).addTo(this.map);
+      
+    this.map.on(L.Draw.Event.CREATED, function(e) {
+      e.layer.feature = {geometry: {type: 'Polygon'} }; 
+      featureGroup.addLayer(e.layer);
+
+      var paramList;
+        $.ajax({
+            type: 'POST',
+            async: false,
+            url: '/api/projects/4/tasks/7/volume',
+            data: JSON.stringify(e.layer.toGeoJSON()),
+            contentType: "application/json",
+            success: function (msg) {
+                paramList = msg;
+            },
+            error: function (jqXHR, textStatus, errorThrown) {
+                alert("get session failed " + errorThrown);
+            }
+        });
+
+      e.layer.bindPopup('Volume: ' + paramList.toFixed(2) + 'Mètres Cubes (m3)');
+    });
+
+    this.map.on(L.Draw.Event.EDITED, function(e) {
+      e.layers.eachLayer(function(layer) {
+        var paramList = null;
+        $.ajax({
+            type: 'POST',
+            async: false,
+            url: '/api/projects/1/tasks/4/volume',
```

Comment:
Project ID and task ID are hard-coded here, perhaps replace them with:

`
const meta = layer[Symbol.for("meta")];
meta.task.project --> project id 
meta.task.id --> task id
`

Reply:
right! i wwould test this,
const meta = layer[Symbol.for("meta")]; 
meta.task.project=project.id 
meta.task.id=task.id